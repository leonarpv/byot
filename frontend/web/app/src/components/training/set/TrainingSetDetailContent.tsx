import * as React from 'react';
import {ControlPanelTitle} from '../../control-panel/base/ControlPanelTitle/ControlPanelTitle';
import {TrainingList} from '../training/TrainingList';
import {TrainingCreateListItem} from '../training/TrainingCreateListItem';
import {ITrainingSet} from '@byot-frontend/common/src/types/interfaces/ITrainingSet';
import {EditConfirmDeleteControls} from '../../elements/controls/EditConfirmDeleteControls';
import {Router} from '../../../router/Router';
import {useTranslation} from 'react-i18next';

interface Props {
  trainingSet?: ITrainingSet;
  onRemove: () => void;
  isRemoving?: boolean;
}

export const TrainingSetDetailContent: React.FC<Props> = (props: Props) => {
  const {t} = useTranslation();

  return (
    <>
      <EditConfirmDeleteControls
        editUrl={Router.training.trainingSet.edit.URI({trainingSetId: props.trainingSet?.id})}
        onDeleteClick={props.onRemove}
        confirmationText={t('Do you really want to delete this training set?')}
        isRemoving={props.isRemoving}
      />
      <ControlPanelTitle>{props.trainingSet?.label}</ControlPanelTitle>
      <TrainingList items={props.trainingSet?.trainings.entries || []}>
        <TrainingCreateListItem />
      </TrainingList>
    </>
  );
};
