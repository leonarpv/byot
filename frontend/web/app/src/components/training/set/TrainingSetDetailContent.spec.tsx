import * as React from 'react';
import {render} from '@testing-library/react';
import {TrainingSetDetailContent} from './TrainingSetDetailContent';

describe('<TrainingSetDetailContent/>', () => {
  it('should render', () => {
    const {container} = render(<TrainingSetDetailContent onRemove={jest.fn()} />);
    expect(container).toMatchSnapshot();
  });
});
