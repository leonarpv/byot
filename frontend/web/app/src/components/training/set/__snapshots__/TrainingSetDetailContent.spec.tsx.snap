// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrainingSetDetailContent/> should render 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-justify-xs-flex-end"
  >
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
        data-testid="app-elements-controls-edit"
        tabindex="0"
        title="Edit"
        to="/training/set/:trainingSetId/edit"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 17.46v3.04c0 .28.22.5.5.5h3.04c.13 0 .26-.05.35-.15L17.81 9.94l-3.75-3.75L3.15 17.1c-.1.1-.15.22-.15.36zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item"
    >
      <button
        class="MuiButtonBase-root MuiIconButton-root makeStyles-remove-2"
        data-testid="app-elements-controls-remove"
        tabindex="0"
        title="Delete"
        type="button"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v10zM18 4h-2.5l-.71-.71c-.18-.18-.44-.29-.7-.29H9.91c-.26 0-.52.11-.7.29L8.5 4H6c-.55 0-1 .45-1 1s.45 1 1 1h12c.55 0 1-.45 1-1s-.45-1-1-1z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
  <div
    class="MuiBox-root MuiBox-root-3"
    data-testid="app-controlPanel-title"
  >
    <h3
      class="MuiTypography-root MuiTypography-h3"
    />
  </div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6"
    >
      <div
        aria-disabled="false"
        class="MuiBox-root MuiBox-root-12 MuiButtonBase-root MuiListItem-root makeStyles-root-4 MuiListItem-gutters MuiListItem-button makeStyles-root-11"
        role="button"
        tabindex="0"
        to="/training/set/:trainingSetId/create-training"
      >
        <div
          class="makeStyles-image-5 makeStyles-image-8"
        >
          <div
            class="makeStyles-root-13"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 13h-5v5c0 .55-.45 1-1 1s-1-.45-1-1v-5H6c-.55 0-1-.45-1-1s.45-1 1-1h5V6c0-.55.45-1 1-1s1 .45 1 1v5h5c.55 0 1 .45 1 1s-.45 1-1 1z"
              />
            </svg>
          </div>
        </div>
        <div
          class="MuiListItemText-root makeStyles-textRoot-7 makeStyles-textRoot-10"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 MuiTypography-colorPrimary"
          >
            Create training
          </h6>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
  </div>
</div>
`;
